maestro-cli/src/main/java/maestro/cli/device/ios/Simctl.kt

```diff
@@ -0,0 +1,50 @@
+package maestro.cli.device.ios
+
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import maestro.MaestroTimer
+import maestro.cli.CliError
+import maestro.cli.util.CommandLineUtils.runCommand
+
+object Simctl {
+    fun list(): SimctlList {
+        val command = listOf("xcrun", "simctl", "list", "-j")
+
+        val process = ProcessBuilder(command).start()
+        val json = String(process.inputStream.readAllBytes())
+
+        return jacksonObjectMapper().readValue(json)
+    }
+
+    fun awaitLaunch(deviceId: String) {
+        MaestroTimer.withTimeout(30000) {
+            if (list()
+                    .devices
+                    .values
+                    .flatMap { it }
+                    .find { it.udid == deviceId }
+                    ?.state == "Booted"
+            ) true else null
+        } ?: throw CliError("Device $deviceId did not boot in time")
+    }
+
+    fun launchSimulator(deviceId: String) {
+        runCommand("xcrun simctl boot $deviceId")
```

Comment:
```suggestion
        runCommand("idb_companion --boot $deviceId")
```

`idb_companion` terminates only after sim is booted up

Reply:
Example output of the command as it's booting up the sim. We could also parse last line and assert whether the sim got booted up successfully or not.

```
idb_companion --boot 726166E0-3766-459B-AAEB-971C0E1D326B

IDB Companion Built at Aug 12 2022 08:41:50
IDB Companion architecture arm64
Booting 726166E0-3766-459B-AAEB-971C0E1D326B
CoreSimulator: Loading from /Library/Developer/PrivateFrameworks/CoreSimulator.framework
CoreSimulator: Successfully loaded
CoreSimulator: SimDevice has correct path of /Library/Developer
AccessibilityPlatformTranslation: Already loaded, skipping
AccessibilityPlatformTranslation: AXPTranslationObject has correct path of /System/Library
Loaded All Private Frameworks [CoreSimulator, AccessibilityPlatformTranslation]
Booting 726166E0-3766-459B-AAEB-971C0E1D326B normally
Booting 726166E0-3766-459B-AAEB-971C0E1D326B with verification
[simulator_set] boot: called with: [Boot Environment {} | Options []]
[simulator_set] disconnectWithTimeout:logger: called with: [30.000000, <FBIDBLogger: 0x6000013a0060>]
[726166E0-3766-459B-AAEB-971C0E1D326B] Simulator connections torn down in 0.000134 seconds
[simulator_set] disconnectWithTimeout:logger: succeeded
[simulator_set] disconnectWithTimeout:logger: called with: [30.000000, <FBIDBLogger: 0x6000013a0060>]
[726166E0-3766-459B-AAEB-971C0E1D326B] Simulator connections torn down in 0.000126 seconds
[simulator_set] disconnectWithTimeout:logger: succeeded
[simulator_set] resolveState: called with: [3]
[simulator_set] resolveState: succeeded
[726166E0-3766-459B-AAEB-971C0E1D326B] Boot Status Changed: Booting | Elapsed 0.000000
[726166E0-3766-459B-AAEB-971C0E1D326B] Boot Status Changed: WaitingOnSystemApp | Elapsed 1.000000
[726166E0-3766-459B-AAEB-971C0E1D326B] Boot Status has not changed from 'WaitingOnSystemApp | Elapsed 1.000000' for 1.578630 seconds
[726166E0-3766-459B-AAEB-971C0E1D326B] Boot Status Changed: WaitingOnSystemApp | Elapsed 3.000000
[726166E0-3766-459B-AAEB-971C0E1D326B] Boot Status Changed: WaitingOnSystemApp | Elapsed 3.000000
[726166E0-3766-459B-AAEB-971C0E1D326B] Boot Status has not changed from 'WaitingOnSystemApp | Elapsed 3.000000' for 1.590303 seconds
[726166E0-3766-459B-AAEB-971C0E1D326B] Boot Status Changed: WaitingOnSystemApp | Elapsed 5.000000
[726166E0-3766-459B-AAEB-971C0E1D326B] Boot Status has not changed from 'WaitingOnSystemApp | Elapsed 5.000000' for 1.564724 seconds
[726166E0-3766-459B-AAEB-971C0E1D326B] Boot Status Changed: Finished | Elapsed 8.000000
[simulator_set] boot: succeeded
{"model":"iPhone 13 mini","os_version":"iOS 16.0","udid":"726166E0-3766-459B-AAEB-971C0E1D326B","architecture":"x86_64","type":"Simulator","name":"iPhone 13 mini","state":"Booted"}
```

Reply:
Thanks for the suggestion. Though that is valid, we are trying to minimise our dependency on `idb_companion` and get by with Apple-provided solutions wherever we can.

Reply:
Sure, I've assumed that if you have to use `idb_companion` anyway, it's worth using it + you don't have to have your own workaround waiting for the sim device to be booted up.